core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -9075,4 +9075,22 @@ public Sql schema(CalciteAssert.SchemaSpec schemaSpec) {
       .withBigQuery()
       .ok(expectedBQSql);
   }
+
+  @Test public void testhashbucket() {
+    final RelBuilder builder = relBuilder();
+    final RexNode formatDateRexNode = builder.call(SqlLibraryOperators.HASHBUCKET,
+        builder.call(SqlLibraryOperators.FARM_FINGERPRINT, builder.scan("EMP").field(0)));
```

Comment:
I have checked, FARM_FINGERPRINT is a bq specific function, why you have added that in source query

Reply:
Done